cmd/apply.go

```diff
@@ -39,6 +39,13 @@ apps, mh acts on all apps in your mh config.`,
 			PrintRendered: viper.GetBool("printRendered"),
 		}
 
+		// Add values passed via CLI using --set
+		if viper.GetStringSlice("set") != nil {
+			envCLIConfig = lib.MHConfig{
```

Comment:
You are overwriting `envCLIConfig` instead of extending it.

Reply:
Also, the condition is unnecessary here, `GetStringSlice` will return an empty slice if no parameters are given on the command line. It will in never return `nil`.